src/App.tsx

```diff
@@ -27,12 +49,34 @@ class App extends Component<any, AppState> {
 
   generateRandomState() {
     let randNum = Math.floor(Math.random() * 1084);
-    getImage(randNum).then((result) => this.setState({ image: result }));
+    getImage(randNum)
+      .then((result) => {
+        this.setState({ image: result });
+      })
+      .catch((error) => {
+        this.setState({ image: `Error loading image: ${error.toString()}` });
```

Comment:
I think instead of updating the image state as a string of "Error..." it would be better to store this value in state with the key of "error" and update state with different error messages depending on the condition. 

We want the image state to represent either having an image or not having an image. This is kind of over-writing that with your error string.